usr/src/uts/common/brand/lx/syscall/lx_socket.c

```diff
@@ -3951,6 +3984,42 @@ lx_getsockopt_tcp(sonode_t *so, int optname, void *optval, socklen_t *optlen)
 		}
 		*optlen = sizeof (int);
 		goto out;
+
+	case LX_TCP_INFO:
+		/*
+		 * We only try to fill in the fields, that we know
+		 * some applications expect.
+		 */
+		if (*optlen < sizeof (lx_tcp_info_t)) {
+			error = EINVAL;
+		} else {
+			*optlen = sizeof (lx_tcp_info_t);
+			bzero(optval, *optlen);
+
+			if (so->so_type != SOCK_STREAM)
+			   goto out; 
+
+			lx_tcp_info_t *ti = (lx_tcp_info_t *)optval;
+			conn_t *con = (struct conn_s *)so->so_proto_handle;
+			tcp_t *tp = con->conn_tcp;
+			ti->tcpi_state = STOL_TCPSTATE(tp->tcp_state);
+			ti->tcpi_rto = tp->tcp_rto_initial;
+			ti->tcpi_last_data_recv = tp->tcp_ts_recent;
+			ti->tcpi_rtt = NSEC2USEC(tp->tcp_rtt_sa);
+			ti->tcpi_rttvar = NSEC2USEC(tp->tcp_rtt_sd);
+			ti->tcpi_snd_ssthresh = tp->tcp_swnd;
+			ti->tcpi_snd_cwnd =  tp->tcp_cwnd;
+			ti->tcpi_snd_mss = tp->tcp_mss;
+			ti->tcpi_unacked = tp->tcp_suna;
+			ti->tcpi_sacked = tp->tcp_rack_cnt;
+			ti->tcpi_pmtu = tp->tcp_initial_pmtu;
+			ti->tcpi_total_retrans = tp->tcp_suna;
```

Comment:
Is this right?

It *appears* that Linux might keep per-connection equivalents of the MIB for tcpRetransSegs bumps.  We don't today.  Remove tcpi_total_retrans please?  Or explain why tcp_suna will work here?

Reply:
I removed the field, as we don't have it right now.